File: FilFeed/src/main/java/com/phicdy/filfeed/ui/MyProgressDialogFragment.java
package com.phicdy.filfeed.ui;

import android.app.Dialog;
import android.app.DialogFragment;
import android.app.ProgressDialog;
import android.os.Bundle;

public class MyProgressDialogFragment extends DialogFragment{
    private static ProgressDialog progressDialog = null;
    private static final String MESSAGE = "message";

    public MyProgressDialogFragment(){}

    public static MyProgressDialogFragment newInstance(String message){
        MyProgressDialogFragment instance = new MyProgressDialogFragment();

        Bundle arguments = new Bundle();
        arguments.putString(MESSAGE, message);

        instance.setArguments(arguments);

        return instance;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState){
        if (progressDialog != null)
            return progressDialog;

        String message = getArguments().getString(MESSAGE);

        progressDialog = new ProgressDialog(getActivity());
        progressDialog.setMessage(message);
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
        setCancelable(false);

        return progressDialog;
    }

    @Override
    public Dialog getDialog(){
        return progressDialog;
    }

    @Override
    public void onDestroy(){
        super.onDestroy();
        progressDialog = null;
    }
}

File: feature_add_curation/src/main/java/com/phicdy/mycuration/feature/addcuration/MyProgressDialogFragment.kt
package com.phicdy.mycuration.feature.addcuration

import android.app.Dialog
import android.app.ProgressDialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment

class MyProgressDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        if (progressDialog != null) return progressDialog as ProgressDialog

        val message = arguments?.getString(MESSAGE) ?: ""
        return ProgressDialog(activity).apply {
            setMessage(message)
            setProgressStyle(ProgressDialog.STYLE_SPINNER)
            isCancelable = false
        }
    }

    override fun getDialog(): Dialog? {
        return progressDialog
    }

    override fun onDestroy() {
        super.onDestroy()
        progressDialog = null
    }

    companion object {
        private var progressDialog: ProgressDialog? = null
        private const val MESSAGE = "message"

        fun newInstance(message: String): MyProgressDialogFragment {
            val instance = MyProgressDialogFragment()
            val arguments = Bundle().apply {
                putString(MESSAGE, message)
            }
            instance.arguments = arguments
            return instance
        }
    }
}
